src/fan.c

```diff
@@ -102,10 +128,11 @@ fan_s *fan_init(unsigned pwm_pin, unsigned pwm_low, unsigned pwm_high, unsigned
 			goto error;
 		}
 		int flags;
+		// It is safer to assume tacho signal requires to be pulled-up than leave the dangling wire
```

Comment:
The compiler checks if I use all possible enum values in the switch, but I don't think anything needs to be changed here. I'm using a very paranoid set of gcc options.